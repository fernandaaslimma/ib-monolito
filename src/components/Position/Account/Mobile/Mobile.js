import React, { Fragment } from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  PrincipalValue,
  SecondaryValue,
  Separator,
  MobileGrid,
  GridInfo,
  Label
} from "../../../../styles/mobileGrid";

import DefaultContent from "../../../common/DefaultContent";
import Icon from "../../../common/Icon";
import { black30 } from "../../../../styles/settings";

import Card from "../../../common/Card";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import ShimmerLoading from "../../../common/ShimmerLoading";

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <MobileGrid columns={2}>
            <GridInfo>
              <Label>{translate("ACCOUNT_NUMBER")}</Label>
              <ShimmerLoading />
            </GridInfo>
            <GridInfo alignRight>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <ShimmerLoading />
            </GridInfo>
          </MobileGrid>
          <Separator />
          <br />
          <MobileRow>
            <Label>{translate("AVAILABLE_BALANCE")}</Label>
            <ShimmerLoading />
          </MobileRow>
          <MobileRow>
            <Label>{translate("BLOCKED_BALANCE")}</Label>
            <ShimmerLoading />
          </MobileRow>
          <MobileRow>
            <Label>{translate("TOTAL_BALANCE")}</Label>
            <ShimmerLoading />
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildAccounts = rows => {
  return rows.map((b, i) => (
    <AssetWrapper
      key={`PositionAccountMobile${i}`}
      data-test={`PositionAccountMobile${i}`}
    >
      <AssetRow>
        <Card>
          <MobileGrid columns={2}>
            <GridInfo>
              <Label>{translate("ACCOUNT_NUMBER")}</Label>
              <SecondaryValue>{b.accountNumber}</SecondaryValue>
            </GridInfo>
            <GridInfo alignRight>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <SecondaryValue>{formatDate(b.date)}</SecondaryValue>
            </GridInfo>
          </MobileGrid>
          <Separator />
          <br />
          <MobileRow>
            <Label>{translate("AVAILABLE_BALANCE")}</Label>
            <PrincipalValue>
              {formatNumber(b.availableBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{translate("BLOCKED_BALANCE")}</Label>
            <PrincipalValue>
              {formatNumber(b.blockedBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{translate("TOTAL_BALANCE")}</Label>
            <PrincipalValue>
              {formatNumber(b.totalBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

function Mobile({ accounts, isEmpty, loading }) {
  return (
    <Fragment>
      <Wrapper>
        {isEmpty && (
          <DefaultContent
            data-test="Empty_Position"
            Icon={() => <Icon type="EmptyWallet" color={black30} />}
            primaryText={translate("NO_RECORDS")}
            secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
          />
        )}
        {!isEmpty &&
          (() => {
            if (loading) {
              return buildShimmerLoadingRow();
            } else {
              return buildAccounts(accounts);
            }
          })()}
      </Wrapper>
    </Fragment>
  );
}

Mobile.defaultProps = {
  accounts: []
};

Mobile.propTypes = {
  accounts: arrayOf(
    shape({
      id: number,
      accountNumber: string,
      date: string,
      totalBalance: number,
      availableBalance: number,
      blockedBalance: number
    })
  ),
  loading: bool.isRequired
};

export default Mobile;
