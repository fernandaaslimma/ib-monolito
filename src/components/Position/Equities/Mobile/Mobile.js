import React, { Fragment } from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  MobileGrid,
  GridInfo,
  InfoWrapper,
  Text,
  TopHeader,
  Info,
  Label,
  TitleHeader,
  HeaderLabel,
  HeaderText,
  HeaderInfo,
  HeaderData,
  Name
} from "../../../../styles/mobileGrid";
import DefaultContent from "../../../common/DefaultContent";
import Card from "../../../common/Card";
import { isValueNullOrUndefined } from "../../../../utils/validations/input";
import { DEFAULT_VALUE } from "../../../../utils/constants";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import ShimmerLoading from "../../../common/ShimmerLoading";

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <InfoWrapper>
            <Name>
              {`${translate("NAME")}: `} <ShimmerLoading />
            </Name>
          </InfoWrapper>
          <MobileRow>
            <Info>
              <Label align="left">{translate("REFERENCE_DATE")}</Label>
              <Text align="left">
                <ShimmerLoading />
              </Text>
            </Info>
            <Info>
              <Label align="right">{`${translate("PORTFOLIO")} (%)`}</Label>
              <Text align="right">
                <ShimmerLoading />
              </Text>
            </Info>
            <Info>
              <Label align="right">{`${translate(
                "GROSS_BALANCE"
              )} (R$)`}</Label>
              <Text align="right">
                <ShimmerLoading />
              </Text>
            </Info>
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildRows = (rows, loading) => {
  if (loading) {
    return buildShimmerLoadingRow();
  }
  return rows.map((b, i) => (
    <AssetWrapper key={`mobileEquity_${i}`}>
      <AssetRow>
        <Card data-test={`Card-${i}`}>
          <InfoWrapper>
            <Name>{`${translate("NAME")}: ${b.name}`}</Name>
          </InfoWrapper>
          <MobileGrid>
            <GridInfo>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <Text>{formatDate(b.date)}</Text>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("PORTFOLIO")} (%)`}</Label>
              <Text>{formatNumber(b.portfolioShare, { digits: 2 })}</Text>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("GROSS_BALANCE")} (R$)`}</Label>
              <Text>{formatNumber(b.grossBalance, { digits: 2 })}</Text>
            </GridInfo>
          </MobileGrid>
          <MobileGrid>
            <GridInfo>
              <Label>{translate("QUANTITY")}</Label>
              <Text>
                {isValueNullOrUndefined(b.quantity)
                  ? DEFAULT_VALUE
                  : formatNumber(b.quantity, { digits: 0 })}
              </Text>
            </GridInfo>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

const buildTopInfo = (totalEquities, loading) => {
  const grossBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalEquities && totalEquities.grossBalance, { digits: 2 })
  );

  return (
    <TopHeader>
      <TitleHeader>{translate("TOTAL")}</TitleHeader>
      <HeaderData>
        <HeaderInfo>
          <HeaderLabel>{`${translate("GROSS_BALANCE")} (R$)`}</HeaderLabel>
          <HeaderText>{grossBalance}</HeaderText>
        </HeaderInfo>
      </HeaderData>
    </TopHeader>
  );
};

function Mobile({ equities, totalEquities, isEmpty, loading }) {
  return (
    <Fragment>
      {isEmpty ? (
        <DefaultContent
          data-test="Empty_Position"
          Icon={() => <Icon type="EmptyWallet" color={black30} />}
          primaryText={translate("NO_RECORDS")}
          secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
        />
      ) : (
        <Wrapper>
          {buildTopInfo(totalEquities, loading)}
          {buildRows(equities, loading)}
        </Wrapper>
      )}
    </Fragment>
  );
}

Mobile.defaultProps = {
  equities: [],
  loading: false,
  isEmpty: false
};

Mobile.propTypes = {
  equities: arrayOf(
    shape({
      name: string,
      date: string,
      issuer: string,
      maturityDate: string,
      grossBalance: number,
      netBalance: number,
      quantity: number,
      portfolioShare: number,
      indexerRate: number,
      indexer: string,
      incomeTaxBalance: number,
      iofBalance: number
    })
  ),
  loading: bool,
  isEmpty: bool
};

export default Mobile;
