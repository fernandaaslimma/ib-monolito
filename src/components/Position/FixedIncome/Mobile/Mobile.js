import React, { Fragment } from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  IncomeName,
  InfoWrapper,
  PrincipalValue,
  SecondaryValue,
  MobileGrid,
  GridInfo,
  Text,
  Label,
  TopHeader,
  TitleHeader,
  HeaderData,
  HeaderInfo,
  HeaderLabel,
  HeaderText
} from "../../../../styles/mobileGrid";
import { IssuanceDate, Issuer } from "./styles";
import { isValueNullOrUndefined } from "../../../../utils/validations/input";
import { CREATE_TRANSACTION, DEFAULT_VALUE } from "../../../../utils/constants";
import DefaultContent from "../../../common/DefaultContent";
import Card from "../../../common/Card";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import ShimmerLoading from "../../../common/ShimmerLoading";
import { AlertMessage, Button } from "react-bocombbm-components";
import { BtnWrapper, Separator, StickyWrapper } from "../../WithdrawLca/styles";
import PendingTransactions from "../../../common/PendingTransactions";
import { checkIfHasAccess } from "../../../common/CanAccess/CanAccess";

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <MobileRow>
            <IncomeName>
              <ShimmerLoading />
            </IncomeName>
            <InfoWrapper>
              <IssuanceDate>{`${translate("ISSUANCE")}:`}</IssuanceDate>
              <Issuer>
                <ShimmerLoading />
              </Issuer>
            </InfoWrapper>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("NET_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              <ShimmerLoading />
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_REVENUE")} (R$)`}</Label>
            <PrincipalValue>
              <ShimmerLoading />
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              <ShimmerLoading />
            </PrincipalValue>
          </MobileRow>
          <Separator />
          <MobileGrid>
            <GridInfo>
              <Label>{translate("EXPIRATION")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{translate("INDEX")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("PORTFOLIO")} (%)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label> {`${translate("INCOME_TAX")} (R$)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("IOF")} (R$)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildRows = (rows, loading) => {
  if (loading) {
    return buildShimmerLoadingRow();
  }
  return rows.map((b, i) => (
    <AssetWrapper key={`mobileFixedIncome_${i}`}>
      <AssetRow>
        <Card data-test={`Card-${i}`}>
          <MobileRow>
            <IncomeName>{b.name}</IncomeName>
            <InfoWrapper>
              <IssuanceDate>{`${translate("ISSUANCE")}: ${formatDate(
                b.issueDate
              )}`}</IssuanceDate>
              <Issuer>{b.issuer}</Issuer>
            </InfoWrapper>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("NET_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.netBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_REVENUE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.grossResultBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.grossBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <Separator />
          <MobileGrid>
            <GridInfo>
              <Label>{translate("MATURITY")}</Label>
              <SecondaryValue>{formatDate(b.maturityDate)}</SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <SecondaryValue>{formatDate(b.date)}</SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{translate("INDEX")}</Label>
              <SecondaryValue>{b.indexer}</SecondaryValue>
            </GridInfo>
          </MobileGrid>
          <MobileGrid>
            <GridInfo>
              <Label>{`${translate("PORTFOLIO")} (%)`}</Label>
              <SecondaryValue>{b.portfolioShare}</SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label> {`${translate("INCOME_TAX")} (R$)`}</Label>
              <SecondaryValue>
                {formatNumber(b.incomeTaxBalance, { digits: 2 })}
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("IOF")} (R$)`}</Label>
              <SecondaryValue>
                {formatNumber(b.iofBalance, { digits: 2 })}
              </SecondaryValue>
            </GridInfo>
          </MobileGrid>
          <MobileGrid>
            <GridInfo>
              <Label>{translate("QUANTITY")}</Label>
              <Text>
                {isValueNullOrUndefined(b.quantity)
                  ? DEFAULT_VALUE
                  : formatNumber(b.quantity, { digits: 2 })}
              </Text>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("RATE")} (%)`}</Label>
              <SecondaryValue>{b.indexerRate}</SecondaryValue>
            </GridInfo>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

const buildTopInfo = (totalFixedIncome, loading) => {
  const portfolioShare = loading ? (
    <ShimmerLoading />
  ) : (
    totalFixedIncome && totalFixedIncome.portfolioShare
  );

  const grossBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalFixedIncome && totalFixedIncome.grossBalance, {
      digits: 2
    })
  );

  const netBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalFixedIncome && totalFixedIncome.netBalance, {
      digits: 2
    })
  );

  const incomeTaxBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalFixedIncome && totalFixedIncome.incomeTaxBalance, {
      digits: 2
    })
  );

  const iofBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalFixedIncome && totalFixedIncome.iofBalance, {
      digits: 2
    })
  );

  const grossResultBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalFixedIncome && totalFixedIncome.grossResultBalance, {
      digits: 2
    })
  );

  return (
    <TopHeader>
      <TitleHeader>{translate("TOTAL")}</TitleHeader>
      <HeaderData>
        <HeaderInfo>
          <HeaderLabel>{`${translate("GROSS_BALANCE")} (R$)`}</HeaderLabel>
          <HeaderText>{grossBalance}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("NET_BALANCE")} (R$)`}</HeaderLabel>
          <HeaderText>{netBalance}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("PORTFOLIO")} (%)`}</HeaderLabel>
          <HeaderText>{portfolioShare}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("INCOME_TAX")} (R$)`}</HeaderLabel>
          <HeaderText>{incomeTaxBalance}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("IOF")} (R$)`}</HeaderLabel>
          <HeaderText>{iofBalance}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("GROSS_REVENUE")} (R$)`}</HeaderLabel>
          <HeaderText>{grossResultBalance}</HeaderText>
        </HeaderInfo>
      </HeaderData>
    </TopHeader>
  );
};

function Mobile({
  fixedIncome,
  totalFixedIncome,
  isEmpty,
  loading,
  pendingTransactions,
  totalLca,
  verifyOnGoingTransactions,
  userInfo
}) {
  return (
    <Fragment>
      {isEmpty ? (
        <DefaultContent
          data-test="Empty_Position"
          Icon={() => <Icon type="EmptyWallet" color={black30} />}
          primaryText={translate("NO_RECORDS")}
          secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
        />
      ) : (
        <Wrapper>
          {buildTopInfo(totalFixedIncome, loading)}
          {checkIfHasAccess(userInfo, [CREATE_TRANSACTION]) &&
            pendingTransactions &&
            pendingTransactions.length > 0 && (
              <PendingTransactions
                backgroundColor={"#EEF1F3"}
                pendingTransactions={pendingTransactions}
                mode={"mobile"}
              />
            )}
          {buildRows(fixedIncome, loading)}
        </Wrapper>
      )}
      {checkIfHasAccess(userInfo, [CREATE_TRANSACTION]) &&
      !loading &&
      (!totalLca || totalLca === 0) ? (
        <AlertMessage
          icon="Attention"
          type="neutral"
          spacing={{
            top: "s",
            bottom: "s",
            right: "s",
            left: "s"
          }}
        >
          {translate("NO_LCA_WITH_LIQUIDITY")}
        </AlertMessage>
      ) : null}

      {checkIfHasAccess(userInfo, [CREATE_TRANSACTION]) ? (
        <StickyWrapper>
          <Separator />
          <BtnWrapper>
            <Button
              disabled={!totalLca || totalLca === 0}
              spacing={{ top: "s", bottom: "s", right: "s", left: "s" }}
              onClick={() => verifyOnGoingTransactions()}
            >
              {translate("FIXED_INCOME_WITHDRAW_LCA")}
            </Button>
          </BtnWrapper>
        </StickyWrapper>
      ) : null}
    </Fragment>
  );
}

Mobile.defaultProps = {
  fixedIncome: [],
  loading: false,
  isEmpty: false
};

Mobile.propTypes = {
  fixedIncome: arrayOf(
    shape({
      name: string,
      date: string,
      issuer: string,
      maturityDate: string,
      grossBalance: number,
      netBalance: number,
      portfolioShare: number,
      indexerRate: number,
      indexer: string,
      incomeTaxBalance: number,
      iofBalance: number,
      quantity: number
    })
  ),
  loading: bool,
  isEmpty: bool
};

export default Mobile;
