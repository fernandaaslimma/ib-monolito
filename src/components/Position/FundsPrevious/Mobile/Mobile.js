import React, { Fragment } from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  InfoWrapper,
  PrincipalValue,
  SecondaryValue,
  Separator,
  MobileGrid,
  GridInfo,
  Label,
  Text,
  TopHeader,
  TitleHeader,
  HeaderData,
  HeaderInfo,
  HeaderLabel,
  HeaderText,
  Name
} from "../../../../styles/mobileGrid";
import DefaultContent from "../../../common/DefaultContent";
import { isValueNullOrUndefined } from "../../../../utils/validations/input";
import { DEFAULT_VALUE } from "../../../../utils/constants";
import Card from "../../../common/Card";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import ShimmerLoading from "../../../common/ShimmerLoading";
import LinkCard from "../../../common/LinkCard";

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <MobileRow>
            <InfoWrapper>
              <Name>
                {`${translate("NAME")}: `}
                <ShimmerLoading />
              </Name>
            </InfoWrapper>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("NET_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              <ShimmerLoading />
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_REVENUE")} (R$)`}</Label>
            <PrincipalValue>
              <ShimmerLoading />
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              <ShimmerLoading />
            </PrincipalValue>
          </MobileRow>
          <Separator />
          <MobileGrid>
            <GridInfo>
              <Label>{translate("APPL_DATE")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("IOF")} (R$)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label> {`${translate("INCOME_TAX")} (R$)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildRows = (rows, loading) => {
  if (loading) {
    return buildShimmerLoadingRow();
  }
  return rows.map((b, i) => (
    <AssetWrapper key={`mobileFunds_${i}`}>
      <AssetRow>
        <Card data-test={`Card-${i}`}>
          <MobileRow>
            <InfoWrapper>
              <Name>{`${translate("NAME")}: ${b.name}`}</Name>
            </InfoWrapper>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("NET_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.netBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_REVENUE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.grossResultBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("GROSS_BALANCE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.grossBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <Separator />
          <MobileGrid>
            <GridInfo>
              <Label>{translate("APPL_DATE")}</Label>
              <SecondaryValue>{formatDate(b.investmentDate)}</SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <SecondaryValue>{formatDate(b.date)}</SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("IOF")} (R$)`}</Label>
              <SecondaryValue>
                {formatNumber(b.iofBalance, { digits: 2 })}
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label> {`${translate("INCOME_TAX")} (R$)`}</Label>
              <SecondaryValue>
                {formatNumber(b.incomeTaxBalance, { digits: 2 })}
              </SecondaryValue>
            </GridInfo>
            <GridInfo>
              <Label>{translate("QUANTITY")}</Label>
              <Text>
                {isValueNullOrUndefined(b.quantity)
                  ? DEFAULT_VALUE
                  : formatNumber(b.quantity, { digits: 8 })}
              </Text>
            </GridInfo>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

const buildTopInfo = (totalFunds, loading) => {
  const portfolioShare = loading ? (
    <ShimmerLoading />
  ) : (
    totalFunds && totalFunds.portfolioShare
  );

  const grossBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalFunds && totalFunds.grossBalance, { digits: 2 })
  );

  const netBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalFunds && totalFunds.netBalance, { digits: 2 })
  );

  return (
    <TopHeader>
      <TitleHeader>{translate("TOTAL")}</TitleHeader>
      <HeaderData>
        <HeaderInfo>
          <HeaderLabel>{`${translate("PORTFOLIO")} (%)`}</HeaderLabel>
          <HeaderText>{portfolioShare}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("GROSS_BALANCE")} (R$)`}</HeaderLabel>
          <HeaderText>{grossBalance}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("NET_BALANCE")} (R$)`}</HeaderLabel>
          <HeaderText>{netBalance}</HeaderText>
        </HeaderInfo>
      </HeaderData>
    </TopHeader>
  );
};

function Mobile({ funds, totalFunds, isEmpty, loading }) {
  return (
    <Fragment>
      {isEmpty ? (
        <DefaultContent
          data-test="Empty_Position"
          Icon={() => <Icon type="EmptyWallet" color={black30} />}
          primaryText={translate("NO_RECORDS")}
          secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
        />
      ) : (
        <Wrapper>
          {buildTopInfo(totalFunds, loading)}
          {buildRows(funds, loading)}
          <AssetWrapper>
            {!loading && (
              <LinkCard
                iconType="FiLoader"
                to="/investments/positions/funds"
                anchorText={translate("CLICK_TO_VIEW_FUNDS")}
                versionText={translate("WISH_TO_SEE_SIMPLIFIED_VERSION")}
                dataTest="LinkToSimplified"
                noSpan
                withUnderline
                fontSize={14}
              />
            )}
          </AssetWrapper>
        </Wrapper>
      )}
    </Fragment>
  );
}

Mobile.defaultProps = {
  funds: [],
  loading: false,
  isEmpty: false
};

Mobile.propTypes = {
  funds: arrayOf(
    shape({
      name: string,
      date: string,
      issuer: string,
      maturityDate: string,
      grossBalance: number,
      netBalance: number,
      portfolioShare: number,
      indexerRate: number,
      indexer: string,
      incomeTaxBalance: number,
      iofBalance: number
    })
  ),
  loading: bool,
  isEmpty: bool
};

export default Mobile;
