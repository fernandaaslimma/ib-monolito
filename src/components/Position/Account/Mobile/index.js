import { default as Mobile } from "./Mobile";

export default Mobile;
