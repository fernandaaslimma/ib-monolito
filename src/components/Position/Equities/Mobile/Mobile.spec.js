import React from "react";
import { shallow } from "enzyme";

import Mobile from "./Mobile";

const equities = [
  {
    name: "name",
    date: "12/12",
    grossBalance: 123,
    portfolioShare: 123
  }
];

const totalEquities = [
  {
    assetType: "Equity",
    assetTypeLabel: "Equity",
    date: "2018-05-04",
    grossBalance: 147547.4,
    netBalance: 147547.4,
    portfolioShare: 4.67
  }
];

describe("Mobile component", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<Mobile equities={equities} totalEquities={totalEquities} />)
    ).toMatchSnapshot();
  });

  it("should match snapshot with loading", () => {
    expect(
      shallow(
        <Mobile equities={equities} totalEquities={totalEquities} loading />
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot with isEmpty", () => {
    expect(
      shallow(
        <Mobile equities={equities} totalEquities={totalEquities} isEmpty />
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot with loading and isEmpty", () => {
    expect(
      shallow(
        <Mobile
          equities={equities}
          totalEquities={totalEquities}
          loading
          isEmpty
        />
      )
    ).toMatchSnapshot();
  });
});
