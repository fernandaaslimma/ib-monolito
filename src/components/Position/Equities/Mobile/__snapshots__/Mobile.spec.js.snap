// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile component should match snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div />
      <styled.div>
        <styled.div>
          <styled.div>
            undefined (R$)
          </styled.div>
          <styled.div>
            NaN
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.section
      key="mobileEquity_0"
    >
      <styled.div>
        <Card
          data-test="Card-0"
          icon={null}
          iconColor={null}
          title={null}
          titleColor={null}
        >
          <styled.div>
            <styled.div>
              undefined: name
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div>
              <styled.div />
              <styled.div>
                06/01/2010
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.div>
                undefined (%)
              </styled.div>
              <styled.div>
                123,00
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.div>
                undefined (R$)
              </styled.div>
              <styled.div>
                123,00
              </styled.div>
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div>
              <styled.div />
              <styled.div>
                ---
              </styled.div>
            </styled.div>
          </styled.div>
        </Card>
      </styled.div>
    </styled.section>
  </styled.div>
</Fragment>
`;

exports[`Mobile component should match snapshot with isEmpty 1`] = `
<Fragment>
  <DefaultContent
    Icon={[Function]}
    data-test="Empty_Position"
    secondaryTexts={
      Array [
        undefined,
      ]
    }
  />
</Fragment>
`;

exports[`Mobile component should match snapshot with loading 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div />
      <styled.div>
        <styled.div>
          <styled.div>
            undefined (R$)
          </styled.div>
          <styled.div>
            <ShimmerLoading
              darker={false}
              index={0}
              inverse={false}
            />
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.section>
      <styled.div>
        <Card
          icon={null}
          iconColor={null}
          title={null}
          titleColor={null}
        >
          <styled.div>
            <styled.div>
              undefined: 
               
              <ShimmerLoading
                darker={false}
                index={0}
                inverse={false}
              />
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div>
              <styled.div
                align="left"
              />
              <styled.div
                align="left"
              >
                <ShimmerLoading
                  darker={false}
                  index={0}
                  inverse={false}
                />
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.div
                align="right"
              >
                undefined (%)
              </styled.div>
              <styled.div
                align="right"
              >
                <ShimmerLoading
                  darker={false}
                  index={0}
                  inverse={false}
                />
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.div
                align="right"
              >
                undefined (R$)
              </styled.div>
              <styled.div
                align="right"
              >
                <ShimmerLoading
                  darker={false}
                  index={0}
                  inverse={false}
                />
              </styled.div>
            </styled.div>
          </styled.div>
        </Card>
      </styled.div>
    </styled.section>
  </styled.div>
</Fragment>
`;

exports[`Mobile component should match snapshot with loading and isEmpty 1`] = `
<Fragment>
  <DefaultContent
    Icon={[Function]}
    data-test="Empty_Position"
    secondaryTexts={
      Array [
        undefined,
      ]
    }
  />
</Fragment>
`;
